### DISCUSSION TITLE
Primefaces11 support in Joinfaces 4.6.x line

### DISCUSSION THREAD

**Discussion by suyambutriveni** [joinfaces/joinfaces#1353] [Q&A]
Hi Team\Lars,

Is there any plan to support Primefaces11 in Joinfaces 4.6.x line?
As per the roadmap of spring framework, I believe that it is going to take a while before the ecosystems switches to Jakarta EE9, considering this scenario, do you think whether it is a good idea to backport changes for Primefaces11 support from the 5.0.x branch to 4.6.x line?

https://spring.io/blog/2021/09/02/a-java-17-and-jakarta-ee-9-baseline-for-spring-framework-6

**Comment by persapiens** [comment]
I agree with @suyambuganesh82 . 

We could have primefaces 11 without jakarta classifier in 4.6.x and primefaces 11 with jakarta classifier in 5.0.x.

**Comment by larsgrefer** [comment]
We definitely should support Primefaces 11 in JoinFaces 4.x.
I'm not sure if we should update to PF 11 in 4.6.x or better wait for 4.7.x
Hiding a major version update of Primefaces in a patch release of JoinFaces might be a breaking change and unexpecting for other users.

**Comment by persapiens** [comment]
@larsgrefer, I appreciate your consideration about breaking change.

If we create the 4.7.x branch because of the Primefaces update, Primefaces should integrate [JoinFaces System Requirements](https://github.com/joinfaces/joinfaces#system-requirements).

If we don't change [JoinFaces System Requirements](https://github.com/joinfaces/joinfaces#system-requirements), we could updade to the 4.6.x branch.

I think we will increase maintenance complexity if we change [JoinFaces System Requirements](https://github.com/joinfaces/joinfaces#system-requirements). That's why I prefer updating the 4.6.x branch.

**Comment by persapiens** [comment]
@larsgrefer, let me reconsider my previous message.

I realized that [Spring Boot 2.7.0-M1](https://github.com/spring-projects/spring-boot/milestone/235) is around the corner. Now I think that we could create JoinFaces 4.7.x branch to follow it with PrimeFaces 11. And we won't need to change Joinfaces System Requirements.

**Reply by larsgrefer** [reply]
I've created the 4.7.x Branch and updated it to SB 2.7.0-M1 and PF 11.
I think we can release a 4.7.0-m1 based on that this weekend
